docs/HACKING.md

```diff
@@ -10,6 +10,17 @@ We welcome all contributions to systemd. If you notice a bug or a missing
 feature, please feel invited to fix it, and submit your work as a GitHub Pull
 Request (PR) at https://github.com/systemd/systemd/pull/new.
 
+systemd uses the meson build system. meson works by generating build files to a
+separate directory (it does not support in-tree builds) and then building from
+that directory. To build systemd in a directory called `build` and run the tests:
+
+```sh
+$ meson setup build
+$ cd build
+$ meson compile
+$ meson test
```

Comment:
```sh
$ meson build
$ meson compile -C build
$ meson test -C build
```

Reply:
Actually, the `compile` command is rather new and not available on Ubuntu Focal yet. I'd just ninja for compilaton/test here then:

```sh
$ meson build
$ ninja -C build
$ ninja -C build test
```

Reply:
regarding `ninja test` vs `meson test`: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=912560